bin/load_exp_design.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -e
+
+scriptDir=$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+source $scriptDir/db_scxa_common.sh
+
+dbConnection=${dbConnection:-$1}
+condensed_sdrf_file=${CONDENSED_SDRF_FILE:-$2}
+sdrf_file=${SDRF_FILE:-$3}
+
+# Check that necessary environment variables are defined.
+[ -z ${dbConnection+x} ] && echo "Env var dbConnection for the database connection needs to be defined. This includes the database name." && exit 1
+[ -z ${CONDENSED_SDRF_FILE+x} ] && echo "Env var CONDENSED_SDRF_FILE for the experiment design data needs to be defined." && exit 1
+[ -z ${SDRF_FILE+x} ] && echo "Env var SDRF_FILE for column sequence of experiment design needs to be defined." && exit 1
+
+# Reason for creating this array is to search factor value column
+# In some sdrf files this column is mentioned as "Factor Value" and in some as "FactorValue"
+ARRAY=( FactorValue "Factor Value" )
+
+while IFS=$'\t' read exp_acc sample sample_type col_name annot_value annot_url
+do
+        if [ $sample_type == 'characteristic' ]
+        then
+            search_column="Characteristics[${col_name}]"
+            column_order=$(awk -v val="$search_column" -F '\t' '{for (i=1; i<=NF; i++) if ($i==val) {print i} }' $sdrf_file)
+        else
+            for element in "${ARRAY[@]}"; do
+              search_column="$element[${col_name}]"
+              column_order=$(awk -v val="$search_column" -F '\t' '{for (i=1; i<=NF; i++) if ($i==val) {print i} }' $sdrf_file)
+            done
```

Comment:
ARRAY will take the values `FactorValue` and `"Factor Value" `.

If in a SDRF file this column is mentioned as "Factor Value", then the second loop in the for (`"Factor Value" `)will produce empty values for `search_column` and `column_order`?

Maybe you should `break` the loop if valid value is found?

Reply:
great catch! added loop break